### ISSUE TITLE
How to use pathprefixstrip and provide path value on Kubernetes Ingress

### ISSUE THREAD

**Issue by padolan** [containous/traefik#863]
I have a tomcat webapp running under kubernetes under a context root of /tomee-sample.  It is deployed to a k8s namespace "us-6000".  I've created a traefik kubernetes ingress using pathprefixstrip, matching "/us-6000/tomee-sample".  The ingress uses the pathprefixstrip annotation, and works great.

My issue is, due to the local context root, I have to duplicate the "tomee-sample" portion of the incoming url to successfully reach the app.  So while "/us-6000/tomee-sample/tomee-sample/healthcheck" works, and resolves to "/tomee-sample/healthcheck", I'd like to be able to reach it through the ingress via "/us-6000/tomee-sample/healthcheck".

Is there a way to specify the pathprefixstrip expression from within the k8s ingress configuration?  I only see examples showing an annotation without a value specified.  I've tried different combinations without success.  Here's my sample ingress yaml that works:
```yml
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  annotations:
    traefik.frontend.rule.type: pathprefixstrip
  name: tomee-sample
  namespace: us-6000
spec:
  rules:
  - http:
      paths:
      - backend:
          serviceName: tomee-sample
          servicePort: 8080
        path: /us-6000/tomee-sample
status:
  loadBalancer: {}
```

I was hoping I could do something as follows:
```yml
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  annotations:
    traefik.frontend.rule.type: pathprefixstrip:/us-6000
  name: tomee-sample
  namespace: us-6000
spec:
  rules:
  - http:
      paths:
      - backend:
          serviceName: tomee-sample
          servicePort: 8080
        path: /us-6000/tomee-sample
status:
  loadBalancer: {}
```
Would appreciate guidance on what I'm doing wrong, or if this feature is not yet supported via k8s ingress.

**Comment by jaygorrell** [comment]
The path that gets stripped is what matches your rule, so changing `path: /us-6000/tomee-sample` to `path: /us-6000` should do what you want. That'll match and strip only the us-6000 portion, then your app will handle the routing of the `/tomee-sample` request.

**Comment by padolan** [comment]
Thanks @jaygorrell ... the problem is, "/us-6000" itself isn't going to be unique enough of a match.  I'll have "/us-6000/tomee-sample", "/us-6000/app1", "/us-6000/app2", etc.

**Comment by jaygorrell** [comment]
Based on your original post:
> I have a tomcat webapp running under kubernetes under a context root of /tomee-sample.

The webserver itself will handle routing that part of the request, you just need to strip off the namespace portion that Traefik handles routing for.

**Comment by padolan** [comment]
These webservers are within individually deployed pods.  e.g. "/app1" will be deployed to us6000-pod-app1, "/app2" will be deployed to us6000-pod-app2, etc.  Traefik is going to be there to take into account both k8s namespace and application to route to the right backend pod.

**Comment by padolan** [comment]
If the design of traefik follows what you mention above:

**Comment by jaygorrell** [comment]
I think it's either that or two tiers of Traefik, but that's bound to get messy. It's not a bad feature request -- possibly an override for the `pathprefixstrip` that each endpoint can define as the piece that gets stripped.

**Comment by ldez** [comment]
Closed by #1638, in the version 1.3.0-rc3.

**Comment by gempesaw** [comment]
Sorry for commenting on a closed thread, but I think I have the same concern as the original issue, and I don't understand how it's solved by #1638.  To refer back to OP's original example, is this functionality available - specifying what to strip, and leaving the rest of the path as is?

```
    traefik.frontend.rule.type: pathprefixstrip:/us-6000
```

I want traefik to redirect requests to `/us-6000/(tomee-sample|something-else)` to my app, and that can be done with an appropriate path setting in the ingress yaml. However, if the path prefix strips the entire matching path, then my app can't tell  whether the request was to `/tomee-sample` or to `/something-else`, as all it sees is a request to `/`, right? 

I found that there's a header with this information #1442, so I guess I can add a conditional middleware that resets the path appropriately before handing it off to the router, but as this is a routing level concern instead of an app concern, if possible I'd like to handle it in the traefik/kubernetes config side.

**Comment by robinvandenberg** [comment]
I was also struggling on this issue. 
It may not really be intended, but I think there is a way to get this to work. 

This is my setup: (traefik 1.5.3)
annotations:
   kubernetes.io/ingress.class: traefik
   traefik.frontend.rule.type: ReplacePathRegex

     - backend:
         serviceName: aServer
         servicePort: 80
       path: /prefix/aServer/(.*) /aServer/$1
     - backend:
         serviceName: aSecondServer
         servicePort: 80
       path: /prefix/aSecondServer/(.*) /aSecondServer/$1

if you use domain/prefix/<the-actual-path>   The backing service will be called with <the-actual-path>.. without the prefix
